tsconfig.json

```diff
@@ -2,7 +2,8 @@
   "extends": "@salesforce/dev-config/tsconfig",
   "compilerOptions": {
     "outDir": "lib",
-    "rootDir": "src"
+    "rootDir": "src",
+    "skipLibCheck": true,
```

Comment:
Is this really needed?
Per the TypeScript [docs](https://www.typescriptlang.org/tsconfig#skipLibCheck) it reduces type system accuracy.

Reply:
@RodEsp there's a dependency of a dependency of a dependency causing the compilation to fail. Setting `skipLibCheck` seems to be the most reasonable approach to me


```
~/repos/salesforcecli/plugin-generate (mdonnalley/update-deps) » yarn compile                               mdonnalley@mdonnal-ltmghj6
yarn run v1.22.19
$ sf-compile
tsc -p . --pretty
node_modules/@types/cacheable-request/index.d.ts:26:42 - error TS2709: Cannot use namespace 'ResponseLike' as a type.

26         cb?: (response: ServerResponse | ResponseLike) => void
                                            ~~~~~~~~~~~~

node_modules/@types/cacheable-request/index.d.ts:77:51 - error TS2709: Cannot use namespace 'ResponseLike' as a type.

77             listener: (response: ServerResponse | ResponseLike) => void
                                                     ~~~~~~~~~~~~

node_modules/@types/cacheable-request/index.d.ts:81:69 - error TS2709: Cannot use namespace 'ResponseLike' as a type.

81         on(event: 'response', listener: (response: ServerResponse | ResponseLike) => void): this;
                                                                       ~~~~~~~~~~~~

node_modules/@types/cacheable-request/index.d.ts:84:71 - error TS2709: Cannot use namespace 'ResponseLike' as a type.

84         once(event: 'response', listener: (response: ServerResponse | ResponseLike) => void): this;
                                                                         ~~~~~~~~~~~~

node_modules/@types/cacheable-request/index.d.ts:89:51 - error TS2709: Cannot use namespace 'ResponseLike' as a type.

89             listener: (response: ServerResponse | ResponseLike) => void
                                                     ~~~~~~~~~~~~

node_modules/@types/cacheable-request/index.d.ts:95:51 - error TS2709: Cannot use namespace 'ResponseLike' as a type.

95             listener: (response: ServerResponse | ResponseLike) => void
                                                     ~~~~~~~~~~~~

node_modules/@types/cacheable-request/index.d.ts:104:51 - error TS2709: Cannot use namespace 'ResponseLike' as a type.

104             listener: (response: ServerResponse | ResponseLike) => void
                                                      ~~~~~~~~~~~~

node_modules/@types/cacheable-request/index.d.ts:108:70 - error TS2709: Cannot use namespace 'ResponseLike' as a type.

108         off(event: 'response', listener: (response: ServerResponse | ResponseLike) => void): this;
                                                                         ~~~~~~~~~~~~

node_modules/@types/cacheable-request/index.d.ts:112:73 - error TS2709: Cannot use namespace 'ResponseLike' as a type.

112         listeners(event: 'response'): Array<(response: ServerResponse | ResponseLike) => void>;
                                                                            ~~~~~~~~~~~~

node_modules/@types/cacheable-request/index.d.ts:115:76 - error TS2709: Cannot use namespace 'ResponseLike' as a type.

115         rawListeners(event: 'response'): Array<(response: ServerResponse | ResponseLike) => void>;
                                                                               ~~~~~~~~~~~~

node_modules/@types/cacheable-request/index.d.ts:118:60 - error TS2709: Cannot use namespace 'ResponseLike' as a type.

118         emit(event: 'response', response: ServerResponse | ResponseLike): boolean;
                                                               ~~~~~~~~~~~~


Found 11 errors in the same file, starting at: node_modules/@types/cacheable-request/index.d.ts:26

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
---------------------------------------------------------------------------------------------------------------------------------------
~/repos/salesforcecli/plugin-generate (mdonnalley/update-deps*) » yarn why @types/cacheable-request/    1 ↵ mdonnalley@mdonnal-ltmghj6
yarn why v1.22.19
[1/4] 🤔  Why do we have the module "@types/cacheable-request/"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "@types/cacheable-request@6.0.2"
info Reasons this module exists
   - "@salesforce#templates#got" depends on it
   - Hoisted from "@salesforce#templates#got#@types#cacheable-request"
info Disk size without dependencies: "92KB"
info Disk size with unique dependencies: "1.95MB"
info Disk size with transitive dependencies: "1.95MB"
info Number of shared dependencies: 4
✨  Done in 0.22s.
```

Reply:
@mdonnalley hmm... I don't get those errors on my machine. Could it be a local env issue?
![image](https://user-images.githubusercontent.com/1084688/188707866-20b5870a-2a5e-4e71-a4ad-74a83d23205c.png)

Reply:
@RodEsp Looks like you're on `main` - have you tried my branch?

Reply:
For posterity: the issue Mike posted above is described [here](https://github.com/sindresorhus/got/issues/2129) and we decided to add `@types/reponse-like` to the yarn resolutions instead of setting `skipLibCheck` to true.